packages/commons-test/src/setupTestContainersVitestGlobal.ts

```diff
@@ -0,0 +1,68 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+/* eslint-disable functional/immutable-data */
+/* eslint-disable functional/no-let */
+
+import { config as dotenv } from "dotenv-flow";
+import { StartedTestContainer } from "testcontainers";
+import {
+  TEST_MINIO_PORT,
+  TEST_MONGO_DB_PORT,
+  TEST_POSTGRES_DB_PORT,
+  TestContainersConfig,
+  minioContainer,
+  mongoDBContainer,
+  postgreSQLContainer,
+} from "./containerTestUtils.js";
+import type {} from "vitest";
+
+declare module "vitest" {
+  export interface ProvidedContext {
+    config: TestContainersConfig;
+  }
+}
```

Comment:
I think there is a way also to properly type provide and inject functions based on this context. I tried to checkout the branch locally and do some experiments adding here in the declare module

```ts
  export type provide = <K extends keyof ProvidedContext>(
    key: K,
    value: ProvidedContext[K]
  ) => void;

  export type inject = <K extends keyof ProvidedContext>(
    key: K
  ) => ProvidedContext[K];
```

But then, when we import inject, it is not typed properly. With provide, here in this file where we do ` provide: (...args: unknown[]) => void;`, it works if I replace it with the type I defined `provide: provide`.

Maybe there is a way to wrap the provide & inject function and re-export them with the new type? Or maybe, should `vitest` do it automatically, and for some reason, it is not working?

In the docs, I see that it says: "You can also extend `ProvidedContext` type to have type safe access to `provide/inject` methods" ( https://vitest.dev/config/#globalsetup )

<img width="483" alt="image" src="https://github.com/pagopa/interop-be-monorepo/assets/6418684/31a14913-b8f0-4b82-a031-6821c4f83f55">

Reply:
Great! Just for consistency, you could move the `type provide` declaration inside the `declare module "vitest" { ... }` like done for `ProvidedContext`, so that it can be imported from `"vitest"` where needed (also in this same file).